### PR TITLE
[quantization] Add ConstProp pass

### PR DESCRIPTION
**Pull request by dayo09**
Let's fold quantized constants too.

### CODE REVIEW

**Review by stamalakhov — APPROVED** [review]
LGTM!

File: tico/utils/convert.py [hunk 1]
```diff
@@ -298,6 +298,7 @@ def convert_exported_module_to_circle(
                 PropagateQParamBackward(),
                 QuantizeBias(),
                 InsertQuantizeOnDtypeMismatch(),
+                ConstPropPass(),
```
[the remark this answers is not in the document]
Comment (stamalakhov) [reply tico/utils/convert.py]: Ahhh... Seems like i missed it. Sorry :cry:

[the remark this answers is not in the document]
Comment (dayo09) [reply tico/utils/convert.py]: @mhs4670go Ah, so the example case I brought up there worked as they only consists of Reshape + Transpose + StridedSlice, so it wasn't the problematic case you have mentioned.

You are worried that quantized input tensors are computed into this logic like below case.

### Constant Folding Logic

To note, below is the ConstProp pass's core logic.
```py
        # propagate constant because all of its args are constant tensors.
        with torch.no_grad():
            prop_constant_tensor = node.target(*args_data, **kwargs_data)
        const_node_to_tensor[node] = prop_constant_tensor 
```


### Quantized Operation's Constant Folding

- reshape.default(quantized_a, shape) : It is folded well.
- add.Tensor(quantized_a, quantized_b) # ERR! : If operation includes calculation, it makes an error as below.

```py
import torch

input_a = torch.quantize_per_tensor(torch.tensor([1.0, 2.0], dtype=torch.float32), 0.1, 10, torch.qint8)
input_b = torch.quantize_per_tensor(torch.tensor([3.0, 4.0], dtype=torch.float32), 0.1, 10, torch.qint8)

target_op_add = torch.ops.aten.add.Tensor
target_op_reshape = torch.ops.aten.reshape.default

try:
    with torch.no_grad():
        result = target_op_add(input_a, input_b) # ERROR!
        # Error: Could not run 'aten::add.out' with arguments from the 'QuantizedCPU' backend. This could be because the operator doesn't exist for this backend, or was omitted during the selective/custom build process (if using custom build). If you are a Facebook employee using PyTorch on mobile, please visit https://fburl.com/ptmfixes for possible resolutions.
        result = target_op_reshape(input_a, torch.tensor([1, 1, 2], dtype=torch.int32)) # PASSES
        print(f"Result Scale: {result.q_scale()}, ZP: {result.q_zero_point()}") 
except Exception as e:
    print(f"Error: {e}")
```

Therefore, the problematic case will be caught by this exception. 
I believe we could implement const props with only allowed shape-relevant operators.

[the remark this answers is not in the document]
Comment (dayo09) [reply tico/utils/convert.py]: @mhs4670go Considering channel-wise parameters could be tricky. Transpose/Permute will be freely performed without changing channel-wise inheritance. But, if `reshape` operator tries to merge `channel(quantized_dim)` axis into other axis, or strided slice tries to split the axis, it could be tricky. Let me create another issue to properly handle diverse cases.
